### DISCUSSION TITLE
isZod

### DISCUSSION THREAD

**Discussion by alii** [colinhacks/zod#720] [Q&A] [answered]
Hi everyone, hope all is well.

I wondered if there was a method to check if something was a zod schema. I'm working with a complex signature right now for something and a value I have could be one of three things, a schema, a function argument or a zod schema. I was looking for something like `z.isZod(something);` but wasn't able to find it.

What would be the best way to do this?

**Comment by digimuza** [comment, marked as answer]
You can easily check 

z.ZodType is base class for all zod schemas so you easily can do
```typescript
const schema = z.object({})
if (schema instanceof z.ZodType) {
// this is zod type
}
```
If you want more specific checks
```typescript
if (schema instanceof z.ZodObject) {
// this is zod object type 
}
```

**Reply by colinhacks** [reply]
If you're looking create a Zod schema for ZodType, you can do this:

```
z.instanceof(z.ZodType);
```

**Reply by MentalGear** [reply]
An official method would be needed, as `obj instanceof z.ZodType` is just not working.

**Reply by eric-poitras** [reply]
@MentalGear is right. I got some cases when using esbuild where the prototype chain gets broken (for some reason I am still investigating) and instanceof is not working. It would be nice to have an official type guard not based on prototypes to validate a zod schema.

@MentalGear How have you resolved your issue? Did you write your own type guard?
